Wrappers/Python/test/test_resample_readers.py

```diff
@@ -59,133 +65,126 @@ def setUp(self):
+            # Using vtk the Y axis gets reversed
+            # vtk_image = Converter.numpy2vtkImage(np.expand_dims(arr[i,:,:], axis=0))
+            # twriter.SetFileName(fnames[-1])
+            # twriter.SetInputData(vtk_image)
+            # twriter.Write()
+            im = Image.fromarray(arr[i])
+            im.save(fnames[-1])
+            
+        self.tiff_fnames = fnames
+
+    def resample_reader_test1(self, reader, target_size):
         # Tests image with correct target size is generated by resample reader:
         # Not a great test, but at least checks the resample reader runs
         # without crashing
         # TODO: improve this test
-        reader = cilRawResampleReader()
+        reader.SetTargetSize(target_size)
+        reader.Modified()
+        reader.Update()
         og_shape = np.shape(self.input_3D_array)
+        raw_type_code = str(self.input_3D_array.dtype)
+        
+
+        if target_size < self.input_3D_array.size * self.bytes_per_element:
+            # Check raw type code was set correctly:
+            self.assertEqual(raw_type_code, reader.GetTypeCodeName())
+            image = reader.GetOutput()
+            extent = image.GetExtent()
+            resulting_shape = (extent[1] + 1, (extent[3] + 1), (extent[5] + 1))
+            og_shape = (og_shape[2], og_shape[1], og_shape[0])
+            og_size = og_shape[0] * og_shape[1] * og_shape[2] * self.bytes_per_element
+            expected_shape = calculate_target_downsample_shape(target_size, og_size, og_shape)
+            self.assertEqual(resulting_shape, expected_shape)
+            # Now test if we get the correct z extent if we set that we
+            # have acquisition data
+            reader.SetIsAcquisitionData(True)
+            # why do we need this?
+            reader.Modified()
+            reader.Update()
+            image = reader.GetOutput()
+            extent2 = image.GetExtent()
+            extent = extent2
+            shape_acquisition = calculate_target_downsample_shape(target_size, og_size, og_shape, acq=True)
+            expected_size = shape_acquisition[0] * \
+                shape_acquisition[1]*shape_acquisition[2]
+            resulting_shape = (extent[1] + 1, (extent[3] + 1), (extent[5] + 1))
+            resulting_size = resulting_shape[0] * \
+                resulting_shape[1]*resulting_shape[2]
+            # angle (z direction) is first index in numpy array, and in cil
+            # but it is the last in vtk.
+            resulting_z_shape = extent[5] + 1
+            og_z_shape = np.shape(self.input_3D_array)[0]
+            self.assertEqual(resulting_size, expected_size)
+            self.assertEqual(resulting_z_shape, og_z_shape)
+        else:
+            reader.Modified()
+            reader.Update()
+            image = reader.GetOutput()
+            resulting_shape = image.GetDimensions()
+            expected_shape = np.shape(self.input_3D_array)
+            # why is this not in the previous test?
```

Comment:
```suggestion
```

it wasn't in the previous test because if we're downsampling, the shape won't be equal to the original shape